src/precompile.c

```diff
@@ -247,85 +240,49 @@ static void _compile_all_union(jl_value_t *sig)
     JL_GC_POP();
 }
 
-static void _compile_all_deq(jl_array_t *found)
-{
-    int found_i, found_l = jl_array_len(found);
-    jl_printf(JL_STDERR, "found %d uncompiled methods for compile-all\n", (int)found_l);
-    jl_method_instance_t *mi = NULL;
-    jl_value_t *src = NULL;
-    JL_GC_PUSH2(&mi, &src);
-    for (found_i = 0; found_i < found_l; found_i++) {
-        if (found_i % (1 + found_l / 300) == 0 || found_i == found_l - 1) // show 300 progress steps, to show progress without overwhelming log files
-            jl_printf(JL_STDERR, " %d / %d\r", found_i + 1, found_l);
-        jl_typemap_entry_t *ml = (jl_typemap_entry_t*)jl_array_ptr_ref(found, found_i);
-        jl_method_t *m = ml->func.method;
-        if (m->source == NULL) // TODO: generic implementations of generated functions
-            continue;
-        mi = jl_get_unspecialized(m);
-        assert(mi == jl_atomic_load_relaxed(&m->unspecialized)); // make sure we didn't get tricked by a generated function, since we can't handle those
-        jl_code_instance_t *ucache = jl_get_method_inferred(mi, (jl_value_t*)jl_any_type, 1, ~(size_t)0);
-        if (jl_atomic_load_relaxed(&ucache->invoke) != NULL)
-            continue;
-        src = m->source;
-        assert(src);
-        // TODO: we could now enable storing inferred function pointers in the `unspecialized` cache
-        //src = jl_type_infer(mi, jl_atomic_load_acquire(&jl_world_counter), 1);
-        //if (jl_atomic_load_relaxed(&ucache->invoke) != NULL)
-        //    continue;
-
-        // first try to create leaf signatures from the signature declaration and compile those
-        _compile_all_union((jl_value_t*)ml->sig);
-        // then also compile the generic fallback
-        jl_generate_fptr_for_unspecialized(ucache);
-    }
-    JL_GC_POP();
-    jl_printf(JL_STDERR, "\n");
-}
-
-static int compile_all_enq__(jl_typemap_entry_t *ml, void *env)
+static int compile_all_collect__(jl_typemap_entry_t *ml, void *env)
 {
-    jl_array_t *found = (jl_array_t*)env;
-    // method definition -- compile template field
+    jl_array_t *allmeths = (jl_array_t*)env;
     jl_method_t *m = ml->func.method;
     if (m->source) {
```

Comment:
Should we try to normalize on whether `!source` or `source == jl_nothing` implies the absence of source code?

Reply:
Yes sounds like a good idea.